### DISCUSSION TITLE
Random Forest: Different behaviors for one class training data between sklearn and intelex

### DISCUSSION THREAD

**Discussion by raychris** [intel/scikit-learn-intelex#1232] [General]
I noticed different behaviors for the random forest classifier implementations between sklearn and intelex for the case of single class training data.  Anyone out there help me understand this one?

In sklearn it seems training data for a multi-class random forest classifier can have a single class:

![image](https://user-images.githubusercontent.com/31640833/229177491-a2486e20-e2d6-4b99-a937-d97f419843dc.png)

When I used the intelex random forest implementation on training data that contained only one class:

File "daal4py/sklearn/ensemble/_forest.py", line 184, in _daal_fit_classifier
    raise ValueError(
ValueError: Training data only contain information about one class.

**Comment by napetrov** [comment]
Interesting question - we might well not support single class classification. @Alexsandruss, @ahuber21 - can you comment?

**Reply by ahuber21** [reply]
I cannot say if there is a strict constraint in the backend that would crash if we set `n_classes=1`, but we are ensuring `n_classes>=2` in multiple places. Isn't this a purely philosophical question? What is the use case of a single-class estimator?

**Reply by raychris** [reply]
I'm coming at it from a practical point of view.  We came across this issue in the workplace so I'll make up an example.  If classifying every star the James Webb Telescope can see was a multi-class problem maybe we have imagery from the telescope and some other inputs in addition to a lot of high quality labeled data from Hubble for training.  Chances are at least a few of the examples of labeled data are single class.  As we're chugging along classifying (at a high rate of speed thanks to intelex) we get some failures due to single class training data.  Why not just go ahead and classify single class in the moment?

**Comment by david-cortes-intel** [comment]
This is now fixed in version 2025.10.

**Reply by raychris** [reply]
Thanks for keeping it on your board and implementing the changes!
